graphistry/plugins/graphviz.py

```diff
@@ -0,0 +1,294 @@
+    nodes_df = positions_df.merge(g._nodes, on=g._node, how='left')
+
+    return g.nodes(nodes_df)
+
+def pgv_styling(g: Plottable) -> Plottable:
+    g2 = g.settings(url_params={
+        'play': 0,
+        'edgeCurvature': 0
+    })
+    return g2
+
+
+def layout_graphviz_core(
+    g: Plottable,
+    prog: Prog = 'dot',
+    args: Optional[str] = None,
+    directed: bool = True,
+    strict: bool = False,
+    graph_attr: Optional[Dict[GraphAttr, Any]] = None,
+    node_attr: Optional[Dict[NodeAttr, Any]] = None,
+    edge_attr: Optional[Dict[EdgeAttr, Any]] = None,
+    drop_unsanitary: bool = False,
+) -> AGraph:
+
+    graph = g_to_pgv(g, directed, strict, drop_unsanitary)
+
+    if graph_attr is not None:
+        for k, v in graph_attr.items():
+            if k not in GRAPH_ATTRS:
+                raise ValueError(f"Unknown graph_attr {k}, expected one of {GRAPH_ATTRS}")
+            if drop_unsanitary and k in UNSANITARY_ATTRS:
+                raise ValueError(f"Unsanitary graph_attr {k} is not allowed")
+            graph.graph_attr[k] = v
+    if node_attr is not None:
+        for k2, v in node_attr.items():
+            if k2 not in NODE_ATTRS:
+                raise ValueError(f"Unknown node_attr {k2}, expected one of {NODE_ATTRS}")
+            if drop_unsanitary and k in UNSANITARY_ATTRS:
+                raise ValueError(f"Unsanitary node_attr {k2} is not allowed")
+            graph.node_attr[k2] = v
+    if edge_attr is not None:
+        for k3, v in edge_attr.items():
+            if k3 not in EDGE_ATTRS:
+                raise ValueError(f"Unknown edge_attr {k3}, expected one of {EDGE_ATTRS}")
+            if drop_unsanitary and k3 in UNSANITARY_ATTRS:
+                raise ValueError(f"Unsanitary edge_attr {k3} is not allowed")
+            graph.edge_attr[k3] = v
+  
+    if prog not in PROGS:
+        raise ValueError(f"Unknown prog {prog}, expected one of {PROGS}")
+
+    if args:
+        #TODO: Security reasoning
+        raise NotImplementedError("NotImplementedError: Passthrough of commandline arguments not implemented")
+
+    graph.layout(prog=prog)
+
+    return graph
+
+
+def layout_graphviz(
+    self: Plottable,
+    prog: Prog = 'dot',
+    args: Optional[str] = None,
+    directed: bool = True,
+    strict: bool = False,
+    graph_attr: Optional[Dict[GraphAttr, Any]] = None,
+    node_attr: Optional[Dict[NodeAttr, Any]] = None,
+    edge_attr: Optional[Dict[EdgeAttr, Any]] = None,
+    skip_styling: bool = False,
+    render_to_disk: bool = False,  # unsafe in server settings
+    path: Optional[str] = None,
+    format: Optional[Format] = None,
+    drop_unsanitary: bool = False,
+) -> Plottable:
+    """
+
+    Use graphviz for layout, such as hierarchical trees and directed acycle graphs
+
+    Requires pygraphviz Python bindings and graphviz native libraries to be installed, see https://pygraphviz.github.io/documentation/stable/install.html
+
+    See PROGS for available layout algorithms
```

Comment:
i know that this is available in the source code but do we want to make `PROGS` and many others (e.g. GRAPH_ATTRS, NODE_ATTRS, etc.) available in the documentation or in the pygraphistry by exposing it?

Reply:
yes